### PR TITLE
Prevent concurrent openwb.conf modifications from mqttsub.py

### PR DESCRIPTION
**Pull request by dj3mu**
According to
https://stackoverflow.com/questions/21936597/blocking-and-non-blocking-subprocess-calls
and
https://docs.python.org/3/library/subprocess.html#subprocess.check_call

subprocess.Popen returns immediately while the called process keeps on running in background.  
Upon quick subsequent MQTT publish this results in multiple processes modifying openwb.conf at same time.  
The last one writing wins and some previous settings might be lost !

Switch to subprocess.run and use of threading.Lock to
prevent concurrent execution of `replaceinconfig.sh`.

Unluckily the diff looks awful as due to introduction of "try/finally" to ensure release of lock, the indentation needed to change.
The main real change is `subprocess.Popen(sendcommand)` is changed to `subprocess.run(sendcommand)`.

### CODE REVIEW

**Comment by dj3mu** [comment]
Ping. Any update on this?

Keeping aligned with head of `master` is not always easy.

Any reason why it's not getting merged?

**Comment by benderl** [comment]
Nothing special, just not enough time for a closer look yet.

File: runs/mqttsub.py [hunk 1]
```diff
@@ -980,6 +980,28 @@ def on_message(client, userdata, msg):
                 if (int(msg.payload) == 2):
                     replaceAll("lademstat=","0")
                     replaceAll("sofortsocstatlp1=","1")
+            if (msg.topic == "openWB/set/lp/2/DirectChargeSubMode"):
```
[the remark this answers is not in the document]
Comment (dj3mu) [reply runs/mqttsub.py:983]: Thanks for careful review! Fully right.  
That's what I meant by previous comment "Keeping align [...] is not always easy"

Should be fixed with commit c03798e495986546298e3798902a5335c97ffa43.

**Comment by benderl** [comment]
Thanks for your patience!

**Pull request merged**